dbcrossbarlib/src/clouds/gcloud/storage/download_file.rs

```diff
@@ -1,27 +1,176 @@
 //! Download a file from Google Cloud storage.
 
+use bytes::BufMut;
+use futures::stream;
+use headers::{ContentRange, Header, HeaderMapExt, Range};
+use reqwest::header::{HeaderMap, HeaderValue, IF_MATCH};
+use std::{cmp::min, convert::TryFrom, ops};
+use tokio::spawn;
+
 use super::{
     super::{percent_encode, AltQuery, Client},
-    parse_gs_url,
+    parse_gs_url, StorageObject, CHUNK_SIZE,
 };
 use crate::common::*;
-use crate::tokio_glue::http_response_stream;
+
+/// Maximum number of parallel downloads.
+const PARALLEL_DOWNLOADS: usize = 5;
```

Comment:
was about to say this should be configurable, but more likely it should just be set by # cores or smth

Reply:
Very little in dbcrossbar cares about number of cores because it's ridiculously async, and only occasionally threaded.

I've hard-coded this because we already allow the user to control the number of _streams_ of 
CSV data. This is the maximum number of simultaneous chunks to download from _each one_ of those CSV streams. So a default of 8 simultaneous streams, and up to 5 chunks per stream.

Allowing the user to dig too deep into these internal, second-level parameters is just too fiddly for my tastes. I'll try to tune reasonable defaults by hand.